### ISSUE TITLE
[explicit-function-return-type] Eslint 6 Problems

### ISSUE THREAD

**Issue by josias-r** [typescript-eslint/typescript-eslint#862]
**Repro**

I upgraded my typescript create-react-app to the latest versions with eslint 6. Therefore I also upgraded to the latest @typescript-eslint versions (2.0.0).

Now my linter reports missing return type on function warnings inside .js files, which shouldn't be the case because those aren't typescript of course.

My eslintrc:
```JSON
{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "react-app",
    "plugin:@typescript-eslint/recommended",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended"
  ],
  "rules": {
    "no-var": 1
  },
  "plugins": ["@typescript-eslint"]
}
```

**Expected Result**

There shouldn't be warnings for js files.

**Actual Result**

I get around 550 warnings for my components now :/

**Additional Info**
I'm using all the recommended settings.
Another error I have now that I've upgraded is the following:
```
Definition for rule '@typescript-eslint/no-angle-bracket-type-assertion' w
as not found  @typescript-eslint/no-angle-bracket-type-assertion
```
I'm not sure if that is related, if not, I'll open another issue for this.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^2.0.0` |
| `@typescript-eslint/parser`        | `^2.0.0` |
| `react-scripts`                       | `^3.1.1` |
| `react-scripts/eslint`(sub dependency)                     | `^6.1.0` |

**Comment by bradzacher** [comment]
As is the standard with semantic versioning - 2.0.0 included a number of breaking changes.

You need to read the release notes so that you can properly address the breaking changes within your project.

https://github.com/typescript-eslint/typescript-eslint/releases/tag/v2.0.0

in particular for you:
- rules are no longer gated to only run on typescript files.
- `no-angle-bracket-type-assertion` has been replaced by another rule.
